lib/prx_access.rb

```diff
@@ -0,0 +1,82 @@
+module PRXAccess
+
+  class PRXHyperResource < HyperResource
+    def incoming_body_filter(hash)
+      super(hash.deep_transform_keys { |key| key.to_s.underscore })
+    end
+
+    class Link < HyperResource::Link
+      attr_accessor :type, :profile
+
+      def initialize(resource, link_spec={})
+        super
+        self.type = link_spec['type']
+        self.profile = link_spec['profile']
+      end
+
+      def where(params)
+        super.tap do |res|
+          res.type = self.type
+          res.profile = self.profile
+        end
+      end
+
+      def headers(*args)
+        super.tap do |res|
+          if args.count > 0
+            res.type = self.type
+            res.profile = self.profile
+          end
+        end
+      end
+    end
+  end
+
+  def default_headers
+    {
+      'Content-Type' => 'application/json',
+      'Accept' =>  'application/json'
+    }
+  end
+
+  def api(options = {})
+    opts = { root: cms_root, headers: default_headers }.merge(options)
+    if account = opts.delete(:account)
+      token = get_account_token(account)
```

Comment:
I'm guessing this doesn't get used often enough to be worth caching across requests.  Right?